src/api/app/models/user.rb

```diff
@@ -129,6 +130,22 @@ def create_home_project
     self.login_failure_count = 0 if login_failure_count.nil?
   end
 
+  def self.autocomplete_login(prefix = '')
+    autocomplete_backend(prefix)
+  end
+
+  def self.autocomplete_token(prefix = '')
+    autocomplete_backend(prefix).collect { |user| { name: user } }
+  end
+
+  def self.autocomplete_backend(prefix = '')
+    if prefix.present?
+      with_login_prefix(prefix).pluck(:login)
+    else
+      User.none
```

Comment:
this return a relation, while in the if part you return an array... I would return `[]` in this case for consistence instead.

Even more, I would write this like:

```ruby
return [] unless prefix.present?
with_login_prefix(prefix).pluck(:login)

Reply:
I already pointed to this in a private conversation :1st_place_medal: